DependencyInjection/CmfBlogExtension.php

```diff
@@ -35,53 +35,57 @@ public function load(array $configs, ContainerBuilder $container)
-        $loader->load('controllers.xml');
+        $loader->load('services.xml');
 
-        if ($config['use_sonata_admin']) {
+        if (isset($config['persistence']['phpcr'])) {
+            $this->loadPhpcrPersistence($config, $loader, $container);
+        }
+
+        if (isset($config['sonata_admin']) && $config['sonata_admin']['enabled']) {
             $this->loadSonataAdmin($config, $loader, $container);
         }
-        if ($config['integrate_menu']['enabled']) {
+
+        if (isset($config['integrate_menu']) && $config['integrate_menu']['enabled']) {
             $this->loadMenuIntegration($config, $loader, $container);
         }
 
-        $config['class'] = array_merge(array(
-            'blog_admin' => 'Symfony\Cmf\Bundle\BlogBundle\Admin\BlogAdmin',
-            'post_admin' => 'Symfony\Cmf\Bundle\BlogBundle\Admin\PostAdmin',
-            'blog' => 'Symfony\Cmf\Bundle\BlogBundle\Document\Blog',
-            'post' => 'Symfony\Cmf\Bundle\BlogBundle\Document\Post',
-        ), isset($config['class']) ? $config['class'] : array());
+        $this->loadPaginationIntegration($config, $container);
+    }
+
+    protected function loadPhpcrPersistence($config, XmlFileLoader $loader, ContainerBuilder $container)
+    {
+        $container->setParameter($this->getAlias().'.blog_basepath', $config['persistence']['phpcr']['blog_basepath']);
 
-        foreach ($config['class'] as $type => $classFqn) {
+        foreach ($config['persistence']['phpcr']['class'] as $type => $classFqn) {
             $container->setParameter(
-                $param = sprintf('cmf_blog.%s_class', $type),
+                $param = sprintf('cmf_blog.phpcr.%s.class', $type),
                 $classFqn
             );
         }
+
+        $loader->load('initializer-phpcr.xml');
+        $loader->load('doctrine-phpcr.xml');
     }
 
-    private function loadSonataAdmin($config, XmlFileLoader $loader, ContainerBuilder $container)
+    protected function loadSonataAdmin(array $config, XmlFileLoader $loader, ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
-        if ('auto' === $config['use_sonata_admin'] && !isset($bundles['SonataDoctrinePHPCRAdminBundle'])) {
+        if (!isset($bundles['SonataDoctrinePHPCRAdminBundle'])) {
             return;
         }
 
-        $loader->load('blog-admin.xml');
-        $container->setParameter($this->getAlias() . '.blog_basepath', $config['blog_basepath']);
+        $loader->load('admin.xml');
     }
 
-    private function loadMenuIntegration($config, XmlFileLoader $loader, ContainerBuilder $container)
+    protected function loadMenuIntegration(array $config, XmlFileLoader $loader, ContainerBuilder $container)
     {
         $bundles = $container->getParameter('kernel.bundles');
-        if ('auto' === $config['integrate_menu']['enabled'] && !isset($bundles['CmfMenuBundle'])) {
+        if (!isset($bundles['CmfMenuBundle'])) {
             return;
```

Comment:
this is removing the true|false|auto mechanism. i add it back in, we
want that for convenient developer experience.